x-pack/plugins/stack_alerts/server/alert_types/es_query/executor.test.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getSearchParams, getValidTimefieldSort, tryToParseAsDate } from './executor';
+import { OnlyEsQueryAlertParams } from './types';
+
+describe('es_query executor', () => {
+  const defaultProps = {
+    size: 3,
+    timeWindowSize: 5,
+    timeWindowUnit: 'm',
+    threshold: [],
+    thresholdComparator: '>=',
+    searchType: 'esQuery',
+    esQuery: '{ "query": "test-query" }',
+    index: ['test-index'],
+    timeField: '',
+  };
+  describe('tryToParseAsDate', () => {
+    it('should parse as date correctly', () => {
+      const expectedResult = '2018-12-31T19:00:00.000Z';
+      expect(expectedResult).toBe(tryToParseAsDate('2019-01-01T00:00:00'));
+      expect(expectedResult).toBe(tryToParseAsDate(1546282800000));
```

Comment:
@Dmitriynj you need to mock Date for this case, else the result may vary, depending on the timezone it is executed